### ISSUE TITLE
New fan control dialog not accessible with screen reader to set fan speed or switch it off

### ISSUE THREAD

**Issue by Nardol** [home-assistant/frontend#16345]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Keyboard navigation looks possible in the list of speeds as I accidentally switched my fan on, but screen reader announces nothing.

### Describe the behavior you expected

When pressing up and down arrow, screen reader should announce which item I selected

### Steps to reproduce the issue

1. Click a fan entity to open the dialog. Using Android I double-tap it, on Linux I use my braille display or the keyboard shortcut to simulate a left click
2. Press tab until the lis has the focus
3. Be sure focus mode is active
4. Press up and down arrow, nothing is announced
5. Try pressing Alt+Down arrow then up/down arrow to be sure the list is expended, result will be the same

### What version of Home Assistant Core has the issue?

2023.4.6

### What was the last working version of Home Assistant Core?

2023.3

### In which browser are you experiencing the issue with?

Firefox 112.0.2, Google Chrome 112.0.5615.165, Home Assistant companion for Android

### Which operating system are you using to run this browser?

Debian Buster, for Android it is 13

**Comment by Nardol** [comment]
Still an issue with Home Assistant 2023.7, I cannot control fan using my keyboard.

**Comment by Nardol** [comment]
I still cannot control my fan speed using keyboard only.
Only thing I can do is switch on or off.

**Comment by Nardol** [comment]
Still an issue, I cannot control fan using a keyboard, same for Android with TalkBack, I'm going to update the issue.

**Comment by Nardol** [comment]
I still can't change fan speed with TalkBack.
On FireFox it looks I can change with arrows then pressing Enter but screen reader does not read what is selected and used.

**Comment by Nardol** [comment]
Still exactly the same, unfortunately.

**Comment by Nardol** [comment]
Still the same

**Comment by Nardol** [comment]
Still an accessibility issue

**Comment by Nardol** [comment]
I've just tested successfully with TalkBack using the Android companion app and Chrome on linux, both radio buttons and lists are accessible. Thanks! 👍
